### PR TITLE
add kb2040 flavor of gherkin

### PR DESCRIPTION
**Pull request by jepler**
.. in order to allow it to be built on the configurator website

## Description

For a possible future Adafruit Learning System Guide, we would like it to be possible to build this firmware via the configurator. In my limited understanding, this is what is needed.

## Types of Changes

I'm not sure of the terminology so I don't know if this is a
- Keyboard (addition or update)

or a
- Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jepler** [comment]
I'll update this and see if I can understand why the ci failed

**Comment by jepler** [comment]
The layout `itsaferbie` failed; the same layout fails for me with essentially the same error if I `make 40percentclub/gherkin:itsaferbie`. Adding `#include <rgblight/rgblight.h>` gets a different error, similar with both versions:
```
jepler@bert:~/src/qmk_firmware$ make 40percentclub/gherkin:itsaferbie  
QMK Firmware 0.19.4
Making 40percentclub/gherkin with keymap itsaferbie

avr-gcc (GCC) 5.4.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiling: quantum/keymap_introspection.c                                                           [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/EndpointStream_AVR8.c                                [OK]
Linking: .build/40percentclub_gherkin_itsaferbie.elf                                                [ERRORS]
 | 
 | .build/obj_40percentclub_gherkin_itsaferbie/quantum/keymap_introspection.o: In function `matrix_init_user':
 | /home/jepler/src/qmk_firmware/quantum/keymap_introspection.c:18: undefined reference to `rgblight_enable'
 | collect2: error: ld returned 1 exit status
 | 
make[1]: *** [builddefs/common_rules.mk:267: .build/40percentclub_gherkin_itsaferbie.elf] Error 1
Make finished with errors
make: *** [Makefile:414: 40percentclub/gherkin:itsaferbie] Error 1
```
I'm not sure how to resolve this problem. As it seems to be a pre-existing problem I hope it's possible to accept this PR without resolving it.

**Comment by jepler** [comment]
The linting failures also seem to exist before this, as well.

**Comment by jepler** [comment]
Hi! I'm still in need of help resolving the build errors that exist prior to this PR...

**Comment by waffle87** [comment]
> Hi! I'm still in need of help resolving the build errors that exist prior to this PR...

You don't need fix/worry about user keymaps that fail without these changes.

**Review by waffle87 — APPROVED** [review]

File: keyboards/40percentclub/gherkin/kb2040/readme.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Gherkin
+
+![Gherkin](https://4.bp.blogspot.com/-sQ18-lNZXOc/WCzlTde-4PI/AAAAAAAB_JQ/qQrehAMG6DMKf3i4oj4mkmLGOfTUvb3KgCLcB/s640/IMG_20161116_122926.jpg)
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/kb2040/readme.md:3]: Please rehost on imgur.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/40percentclub/gherkin/kb2040/readme.md (hunk 1 above)
Comment (jepler) [reply keyboards/40percentclub/gherkin/kb2040/readme.md:3]: I used the same URL already present in the parent readme:
```
keyboards/40percentclub/gherkin/kb2040/readme.md:![Gherkin](https://4.bp.blogspot.com/-sQ18-lNZXOc/WCzlTde-4PI/AAAAAAAB_JQ/qQrehAMG6DMKf3i4oj4mkmLGOfTUvb3KgCLcB/s640/IMG_20161116_122926.jpg)
keyboards/40percentclub/gherkin/readme.md:![Gherkin](https://4.bp.blogspot.com/-sQ18-lNZXOc/WCzlTde-4PI/AAAAAAAB_JQ/qQrehAMG6DMKf3i4oj4mkmLGOfTUvb3KgCLcB/s640/IMG_20161116_122926.jpg)
```

Comment (drashna) [reply keyboards/40percentclub/gherkin/kb2040/readme.md:3]: The original readme is from ~4 years ago, before we added a bunch of the policies.  

Eg, that readme should be fixed too, but would consider that to be outside the scope of this PR.

Comment (jepler) [reply keyboards/40percentclub/gherkin/kb2040/readme.md:3]: I have re-hosted both images on imgur

**Review by drashna — APPROVED** [review]

**Pull request merged**